pkg/meta/meta.go

```diff
@@ -1560,7 +1576,50 @@ func (m *Meta) GetHistoryDDLJobsIterator(startJobID int64) (LastJobIterator, err
 
 // HLastJobIterator is the iterator for gets the latest history.
 type HLastJobIterator struct {
-	iter *structure.ReverseHashIterator
+	iter       *structure.ReverseHashIterator
+	schemaIDs  set.Int64Set
+	tableNames set.StringSet
+}
+
+// extractSchemaIDAndTableName extract schema_id and table_id from encoded Job structure
+// Note, here we strongly rely on the order of fields in marshalled string, just like checkSubstringsInOrder
+func extractSchemaIDAndTableName(s string) (int64, string, error) {
+	pos := strings.Index(s, `"schema_id":`)
+	if pos == -1 {
+		return 0, "", fmt.Errorf("schema_id not found in model.Job json")
+	}
+
+	start := pos + len(`"schema_id":`)
+	substr := s[start:]
```

Comment:
I'm not sure if we should use `[]byte` rather than `string` type because `string` type may create another instance after slicing.

Reply:
seems OK https://stackoverflow.com/questions/52395730/does-slice-of-string-perform-copy-of-underlying-data